### PR TITLE
Optimize performance

### PR DESCRIPTION
**Pull request by liudezhi2098**
Optimize performance

### CODE REVIEW

File: src/main/java/org/apache/pulsar/io/sftp/sink/FileWriteThread.java [hunk 1]
```diff
@@ -32,6 +31,10 @@
  * and publishes them to a Pulsar topic.
  */
 public class FileWriteThread extends Thread {
+    public static final String FILE_NAME = "file.name";
```
Comment (JavaXiaoJun) [inline src/main/java/org/apache/pulsar/io/sftp/sink/FileWriteThread.java:34]: Can use constants defined by org.apache.pulsar.io.sftp.utils.Constants

File: src/main/java/org/apache/pulsar/io/sftp/source/SFTPProcessedThread.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+public class SFTPProcessedThread extends TaskThread {
+
+    private final SFTPSource sftpSource;
+    private final SFTPUtil sftp;
+    private boolean stop = false;
+
+    public SFTPProcessedThread(SFTPSource sftpSource) {
+        this.sftpSource = sftpSource;
+        SFTPSourceConfig sftpConfig = sftpSource.getSFTPSourceConfig();
+        SFTPUtil sftp = new SFTPUtil(sftpConfig.getUsername(), sftpConfig.getPassword(), sftpConfig.getHost(),
```
Comment (JavaXiaoJun) [inline src/main/java/org/apache/pulsar/io/sftp/source/SFTPProcessedThread.java:41]: 线程中所有操作共用一个sftp channel 和 session 可能会在网络波动的情况下导致整个程序中断。创建sftp channel 不是个很重的操作，是否对每个文件都create新的sftp channel更合适？虽然效率上会低一点，但source程序本身会更健壮。

File: src/main/java/org/apache/pulsar/io/sftp/source/SFTPSourceRecord.java [hunk 3]
```diff
@@ -48,12 +50,13 @@ public class SFTPSourceRecord implements Record<byte[]> {
...
 
-    public SFTPSourceRecord(String fileName, byte[] byt, String absolutePath, String modifiedTime)
-            throws NoSuchAlgorithmException, IOException {
+    public SFTPSourceRecord(String fileName, byte[] byt, String absolutePath, String realAbsolutePath,
+                            String modifiedTime) throws NoSuchAlgorithmException, IOException {
         this.key = Optional.of(fileName);
         this.value = byt;
         this.setProperty(FILE_NAME, fileName);
         this.setProperty(FILE_ABSOLUTE_PATH, absolutePath);
+        this.setProperty(Constants.FILE_ABSOLUTE_PATH, StringUtils.isBlank(realAbsolutePath) ? "." : realAbsolutePath);
```
Comment (JavaXiaoJun) [inline src/main/java/org/apache/pulsar/io/sftp/source/SFTPSourceRecord.java:59]: `FILE_ABSOLUTE_PATH` is initialized twice

File: src/main/java/org/apache/pulsar/io/sftp/source/SFTPSourceConfig.java [hunk 4]
```diff
@@ -210,16 +225,17 @@ public void validate() {
...
-        if (StringUtils.isNotBlank(password)) {
-            sftp = new SFTPUtil(username, password, host, port);
-        } else {
-            sftp = new SFTPUtil(username, host, port, privateKey);
-        }
-        sftp.login();
-        Boolean exist = sftp.isDirExist(directory);
-        sftp.logout();
-        return exist;
+//        SFTPUtil sftp;
```
Comment (JavaXiaoJun) [inline src/main/java/org/apache/pulsar/io/sftp/source/SFTPSourceConfig.java:228]: Need to remove code comments

File: src/main/java/org/apache/pulsar/io/sftp/source/SFTPSourceConfig.java [hunk 5]
```diff
@@ -120,6 +124,17 @@ public class SFTPSourceConfig implements Serializable {
      */
     private Integer numWorkers = 1;
 
+    /**
+     * Used to distribute synchronization tasks,
+     * The producer has only one instance listening to the directory through WaitForExclusive mode.
+     */
+    private String sftpTaskTopic = "sftp_task";
```
Comment (JavaXiaoJun) [inline src/main/java/org/apache/pulsar/io/sftp/source/SFTPSourceConfig.java:131]: `sftpTaskTopic` needs to be validated as a non-empty string in the `validate()` function

File: src/main/java/org/apache/pulsar/io/sftp/sink/FileWriteThread.java (hunk 1 above)
Comment (liudezhi2098) [reply src/main/java/org/apache/pulsar/io/sftp/sink/FileWriteThread.java:34]: yes, this needs to be moved out later, I will modify it.

File: src/main/java/org/apache/pulsar/io/sftp/source/SFTPProcessedThread.java (hunk 2 above)
Comment (liudezhi2098) [reply src/main/java/org/apache/pulsar/io/sftp/source/SFTPProcessedThread.java:41]: Establishing a connection for each operation is not a good way to optimize the robustness here.

File: src/main/java/org/apache/pulsar/io/sftp/source/SFTPSourceConfig.java (hunk 5 above)
Comment (JavaXiaoJun) [inline src/main/java/org/apache/pulsar/io/sftp/source/SFTPSourceConfig.java:131]: `sftpTaskTopic` needs to ensure that it does not have the same name as `destination-topic-name` in source create  command

File: src/main/java/org/apache/pulsar/io/sftp/source/SFTPProcessedThread.java (hunk 2 above)
Comment (JavaXiaoJun) [reply src/main/java/org/apache/pulsar/io/sftp/source/SFTPProcessedThread.java:41]: ok, needs to be a better way to optimize robustness later

**Pull request merged**